File: tms-backend/src/main/java/com/tms/dto/request/FleetSnapshotRequest.java
package com.tms.dto.request;

import lombok.Data;

import java.math.BigDecimal;

@Data
public class FleetSnapshotRequest {
    private int totalVehicles;
    private int availableVehicles;
    private int busyVehicles;
    private int maintenanceVehicles;
    private int totalDrivers;
    private int activeDrivers;
    private int inactiveDrivers;
    private int totalTrips;
    private int plannedTrips;
    private int inProgressTrips;
    private int completedTrips;
    private BigDecimal totalExpenses;
    private int totalBookings;
}

File: tms-dotnet-reports/Models/FleetSnapshotRequest.cs
namespace TmsReports.Models;

public class FleetSnapshotRequest
{
    public int TotalVehicles { get; set; }
    public int AvailableVehicles { get; set; }
    public int BusyVehicles { get; set; }
    public int MaintenanceVehicles { get; set; }
    public int TotalDrivers { get; set; }
    public int ActiveDrivers { get; set; }
    public int InactiveDrivers { get; set; }
    public int TotalTrips { get; set; }
    public int PlannedTrips { get; set; }
    public int InProgressTrips { get; set; }
    public int CompletedTrips { get; set; }
    public decimal TotalExpenses { get; set; }
    public int TotalBookings { get; set; }
}
